sdk/keyvault/azure-security-keyvault/azure/security/keyvault/aio/secrets/samples/hello_world_async.py

```diff
@@ -0,0 +1,101 @@
+    # Instantiate a secret client that will be used to call the service.
+    # Notice that the client is using default Azure credentials.
+    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
+    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    credential = AsyncDefaultAzureCredential()
+    client = SecretClient(vault_url="https://vault99e4122d.vault.azure.net", credential=credential)
+    try:
+        # Let's create a secret holding bank account credentials valid for 1 year.
+        # if the secret already exists in the key vault, then a new version of the secret is created.
+        print("\n1. Create Secret")
+        expires = datetime.datetime.now() + datetime.timedelta(days=365)
+        secret = await client.set_secret("secretName", "secretValue", expires=expires)
+        print("Secret with name '{0}' created with value {1}".format(secret.name, secret.value))
+        print("Secret with name '{0}' expires on {1}".format(secret.name, secret.expires))
+
+        # Let's get the bank secret using its name
+        print("\n2. Get a Secret by name")
+        bank_secret = await client.get_secret("secretName")
+        print("Secret with name '{0}' was found with value {1}.".format(bank_secret.name, bank_secret.value))
+
+        # After one year, the bank account is still active, we need to update the expiry time of the secret.
+        # The update method can be used to update the expiry attribute of the secret. It cannot be used to update
+        # the value of the secret.
+        print("\n3. Update a Secret by name")
+        expires = bank_secret.expires + datetime.timedelta(days=365)
+        updated_secret = await client.update_secret("secretName", expires=expires)
+        # To ensure secret is updated on the server side.
+        time.sleep(5)
+        print("Secret with name '{0}' was updated on date {1}".format(updated_secret.name, updated_secret.updated))
+        print("Secret with name '{0}' was updated to expire on {1}".format(updated_secret.name, updated_secret.expires))
+
+        # Bank forced a password update for security purposes. Let's change the value of the secret in the key vault.
+        # To achieve this, we need to create a new version of the secret in the key vault. The update operation cannot
+        # change the value of the secret.
+        new_secret = await client.set_secret("secretName", "newSecretValue")
+        # To ensure secret is updated on the server side.
+        time.sleep(5)
+        print("Secret with name '{0}' created with value {1}".format(new_secret.name, new_secret.value))
+
+        # The bank account was closed, need to delete its credentials from the Key Vault.
+        print("\n4. Delete Secret")
+        print("Deleting Secret..")
+        deleted_secret = await client.delete_secret("secretName")
+        # To ensure secret is deleted on the server side.
+        time.sleep(20)
```

Comment:
Creating a sample just for soft-delete scenarios might be best. Having one place to explain what soft-delete allows means not needing to describe it everywhere else; other samples become shorter in length and execution time, and less likely to fail.

Reply:
Added sample for recover_purge to cover soft-delete case and removed from other samples.